Sources/CognitoIdentityDemo/CognitoIdentityDemo.swift

```diff
@@ -47,66 +54,69 @@ class CognitoIdentityDemo {
                 switch(result) {
                 case .success(let output):
                     if let identityPools = output.identityPools {
-                        for pool in identityPools {
-                            if pool.identityPoolName == name,
-                               let poolId = pool.identityPoolId {
-                                completion(poolId)
-                                break
-                            }
+                        let foundID = identityPools.filter {
+                            $0.identityPoolName == name
+                        }
+                        .map {
+                            $0.identityPoolId
+                        }
+                        .first ?? nil
+                        
+                        if let foundID = foundID {
+                            poolID = foundID
```

Comment:
This can be written more succinctly as:
```
                case .success(let output):
                    poolID = output.identityPools?
                        .filter { $0.identityPoolName == name }
                        .map { $0.identityPoolId }
                        .first ?? nil
                    token = output.nextToken
```

Reply:
I originally wrote something similar but it didn't work right, so eventually I gave up and did it the way shown above. I'm applying your suggestion now.